mathesar/tests/api/test_table_api.py

```diff
@@ -2010,3 +2023,32 @@ def test_create_table_with_nested_json_objects(client, schema):
             client, table_name, table_name, datafile, schema, expected_data[index]["first_row"],
             expected_data[index]["column_names"], import_target_table=None
         )
+
+
+def test_create_table_using_excel_data_file(client, patents_excel_data_file, schema):
+    table_name = 'patents'
+    expt_name = _get_expected_name(table_name, data_file=patents_excel_data_file)
+    first_row = (
+        1,
+        "NASA Kennedy Space Center",
+        "Application",
+        "KSC-12871",
+        "0",
+        "13/033,085",
+        "Polyimide Wire Insulation Repair System",
+        None,
+    )
+    column_names = [
+        "Center",
+        "Status",
+        "Case Number",
+        "Patent Number",
+        "Application SN",
+        "Title",
+        "Patent Expiration Date",
+    ]
+
+    check_create_table_response(
+        client, table_name, expt_name, patents_excel_data_file, schema, first_row,
+        column_names, import_target_table=None
+    )
```

Comment:
Please refactor this and other API tests you wrote (meaning tests within `mathesar/tests/api/`) to import tables via HTTP API calls, not Python calls. Same for querying them. Here, the patents table is created via `DataFile.objects.create` inside the fixture you wrote.

We talked about this before, but the motivation for having HTTP API tests is that we want these tests to integrate as much of the backend stack as possible.